src/components/CheckboxInput/CheckboxInput.tsx

```diff
@@ -31,32 +35,24 @@ export type CheckboxInputProps = CheckboxHTMLElementProps & {
 export const CheckboxInput = React.forwardRef<
   HTMLInputElement,
   CheckboxInputProps
->(({ checked, className, disabled, ...other }, ref) => {
-  // Make indeterminate checkbox visually match the colors of a checked state, but announce itself
-  // as "mixed" to screen readers
-  //
-  // However, https://html.spec.whatwg.org/multipage/input.html#checkbox-state-(type=checkbox)
-  // seems to consider `indeterminate` as a separate bit of information than `checked`. They can be
-  // in all 4 combinations (only 3 visual states, though).
-  //
-  // Should we make them separate props?
-  const checkedProps =
-    checked === 'indeterminate'
-      ? {
-          'aria-checked': 'mixed' as const,
-          checked: true,
-        }
-      : {
-          checked,
-        };
+>(({ checked, className, disabled, indeterminate, ...other }, ref) => {
+  const forwardedRef = useForwardedRef(ref);
+
+  // Make this checkbox indeterminate. Can only be done with JS for some reason.
```

Comment:
See https://github.com/facebook/react/issues/1798 for why React hasn't added an `indeterminate` prop to input checkboxes